File: exercise/php/day22/src/Eid/Sex.php
<?php

namespace Eid;

use Monad\Option;
use Monad\Option\None;
use Monad\Option\Some;

enum Sex: string
{
    case SLOUBI = '1';
    case GAGNA = '2';
    case CATACT = '3';

    public static function parse(string $potentialSex): Option
    {
        return match ($potentialSex) {
            '1' => new Some(self::SLOUBI),
            '2' => new Some(self::GAGNA),
            '3' => new Some(self::CATACT),
            default => new None(new ParsingError("Not a valid sex")),
        };
    }

    public function toString(): string
    {
        return $this->value;
    }
}

File: exercise/kotlin/day22/src/main/kotlin/eid/Sex.kt
package eid

enum class Sex {
    Sloubi, Gagna, Catact;

    override fun toString(): String =
        when (this) {
            Sloubi -> "1"
            Gagna -> "2"
            Catact -> "3"
        }

    companion object {
        fun parse(potentialSex: Char): Sex? =
            when (potentialSex) {
                '1' -> Sloubi
                '2' -> Gagna
                '3' -> Catact
                else -> null
            }
    }
}
